doc/src/vpicUnits.tex

```diff
@@ -37,41 +51,156 @@
+
+	\section{Species \& Particle Weighting}
+
+	To define a particle species in VPIC, the user specifies the species mass $m_{\mathrm{s}}$ and charge $q_{\mathrm{s}}$. Additionally, each particle is individually assigned a weighting factor $w$. These must be consistent with each other and the above unit system to produce the desired behaviour.
+
+	\subsection{Choosing Charge Mass Ratio}
+
+	The motion of a physical particle with charge $q_{\mathrm{p}}$ and mass $m_{\mathrm{p}}$ in the EM fields only has explicit dependence on the charge-mass ratio $r \equiv q_{\mathrm{p}}/m_{\mathrm{p}}$. This is also the case for macroparticles in VPIC's particle push. Therefore the corresponding macroparticle species must be defined such that its charge $q_{\mathrm{s}}$ and mass $m_{\mathrm{s}}$ satisfy $q_{\mathrm{s}}/m_{\mathrm{s}}=r$.
+
+	\subsection{Choosing Macroparticle Weight}
+	When calculating the fields generated by a macroparticle, the code uses the species charge and macroparticle weight to calculate a total charge for the macroparticle:
 
-	Particle momenta are the only user-facing exception to the unit system defined above. The momenta are normalised to produce a dimensionless momentum
-	
 	\begin{equation}
-		u \equiv \frac{p}{mc}.
+		q_{\mathrm{MP}} = wq_{\mathrm{s}}.
 	\end{equation}
+
+	\noindent For this to result in an amount of current deposition consistent with the desired physical particle density, the macroparticle weight should be defined such that the charge contained in volume $V$ is the same for the macroparticles as for physical particles, i.e.
+
+	\begin{align*}
+		q_{\mathrm{p}}n_{\mathrm{p}}V &= q_{\mathrm{MP}}n_{\mathrm{MP}}V \\
+			&= wq_{\mathrm{s}}n_{\mathrm{MP}}V,
+	\end{align*}
+
+	\noindent where $n_{\mathrm{p}}$ and $n_{\mathrm{MP}}$ are the physical particle and macroparticle number densities (macroparticles per unit volume) respectively. So, the particle weight must satisfy
+
+	\begin{align}
+		w = \frac{q_{\mathrm{p}}n_{\mathrm{p}}}{q_{\mathrm{s}}n_{\mathrm{MP}}} = \frac{\rho_{\mathrm{p}}}{\rho_{\mathrm{s}}}.
+	\end{align}
+
+	\noindent This illustrates that there is a degree of freedom available in that the species charge defined in the deck does not have to be the same as the physical species charge.
+
+	\subsection{Consequences of $q_{\mathrm{s}} \neq q_{\mathrm{p}}$}
+
+	The possibility of $q_{\mathrm{s}} \neq q_{\mathrm{p}}$ means that the code has no way of calculating numbers of physical particles since it doesn't know the true particle charge. VPIC has been carefully designed to produce valid output despite this, which for example is why the hydro dumps contain charge density rather than number density. Users should therefore also ensure that any custom diagnostics do not assume $q_{\mathrm{s}} = q_{\mathrm{p}}$.
```

Comment:
I don't think there's any careful design here.  The number density doesn't appear in the equations VPIC solves, so VPIC doesn't know what the number density is.

Reply:
Isn't this kind of splitting hairs?

I don't see a problem with saying it is 'carefully designed' - perhaps it is obvious to you that this is how the code has to be set up but it might not be to other users of the code?